problemtools/verifyproblem.py

```diff
@@ -470,6 +470,14 @@ def check(self, args):
             if not f[:-4] + '.in' in infiles:
                 self.error("No matching input file for answer '%s'" % f)
 
+        # Check that test data filenames are legal (according to the problem
+        # format standard).
+        filename_regex = re.compile('^[a-zA-Z0-9][a-zA-Z0-9_.-]*[a-zA-Z0-9]$')
```

Comment:
There is something grating about this regex being recompiled for every test case group, even though it presumably has no noticeable performance impact.  Shouldn't this be a static/constant defined in a slightly less local scope?